src/engine/client/input.cpp

```diff
@@ -191,6 +228,69 @@ int CInput::Update()
 					Scancode = Event.key.keysym.scancode;
 					break;
 
+				// handle the joystick events
+				case SDL_JOYBUTTONUP:
+					Action = IInput::FLAG_RELEASE;
+
+					// fall through
+				case SDL_JOYBUTTONDOWN:
+					dbg_msg("joystick", "Joystick button down: %d", Event.jbutton.button);
```

Comment:
This (and the `dbg_msg` below) looks quite verbose, right? I don't have a joystick to test but doing a print for every joystick event sounds too much.